#VRML_SIM R2022a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.7038230825987861 -0.6998545525068097 -0.1218058854761765 3.379053989628036
  position 0.041755757812343613 2.5552403461829654 0.9094075603744392
}
TexturedBackground {
  texture "factory"
}
TexturedBackgroundLight {
}
DEF RedCylinder Solid {
  translation 0.5532305139312528 0.14925105434898356 -0.5080005484220264
  rotation 0.9999170433353742 -0.009107870113450763 -0.00910786195766343 1.5708793032283161
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.381
        radius 0.08
      }
    }
  ]
  name "solid(1)"
  boundingObject DEF landmark Shape {
    appearance PBRAppearance {
      baseColor 0 1 1
      roughness 1
      metalness 0
    }
    geometry Cylinder {
      height 0.3
      radius 0.045
    }
  }
  physics DEF landmarkPH Physics {
  }
  locked TRUE
  recognitionColors [
    1 0 0
  ]
}
DEF YellowCylinder Solid {
  translation -0.508 0.149251097485959 -0.5079994507452452
  rotation -1 -4.554592264605575e-16 -1.761166121795832e-16 1.5707963382340193
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.381
        radius 0.08
      }
    }
  ]
  name "solid(4)"
  boundingObject DEF landmark Shape {
    appearance PBRAppearance {
      baseColor 0 1 1
      roughness 1
      metalness 0
    }
    geometry Cylinder {
      height 0.3
      radius 0.045
    }
  }
  physics DEF landmarkPH Physics {
  }
  locked TRUE
  recognitionColors [
    1 1 0
  ]
}
DEF GreenCylinder Solid {
  translation -0.508 0.1492510974859562 0.5079998475368129
  rotation -1 -4.763824699793937e-16 -2.6555369656200205e-16 1.5707963236196025
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.381
        radius 0.08
      }
    }
  ]
  name "solid(3)"
  boundingObject DEF landmark Shape {
    appearance PBRAppearance {
      baseColor 0 1 1
      roughness 1
      metalness 0
    }
    geometry Cylinder {
      height 0.3
      radius 0.045
    }
  }
  physics DEF landmarkPH Physics {
  }
  locked TRUE
  recognitionColors [
    0 1 0
  ]
}
DEF BlueCylinder Solid {
  translation 0.508 0.14925109748595639 0.5079998475368129
  rotation -1 -6.05418337208119e-16 -3.177136717694197e-16 1.5707963236196028
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.381
        radius 0.08
      }
    }
  ]
  name "solid(5)"
  boundingObject DEF landmark Shape {
    appearance PBRAppearance {
      baseColor 0 1 1
      roughness 1
      metalness 0
    }
    geometry Cylinder {
      height 0.3
      radius 0.045
    }
  }
  physics DEF landmarkPH Physics {
  }
  locked TRUE
  recognitionColors [
    0 0 1
  ]
}
DEF area RectangleArena {
  rotation -0.5773502691896258 0.5773502691896258 0.5773502691896258 2.0943951023931957
  floorSize 1.016 1.016
  floorTileSize 0.508 0.508
  wallAppearance BrushedAluminium {
    colorOverride 0 0 0
  }
}
DEF epuck E-puck {
  translation 0.0545255 -8.87461e-05 -0.0979596
  rotation -0.0009560315939150619 0.7065016999056297 0.7077106993920939 3.13815
  controller "lab5task1"
  supervisor TRUE
  camera_width 50
  turretSlot [
    InertialUnit {
    }
    DEF righthinge HingeJoint {
      jointParameters HingeJointParameters {
        position -9.733307049231144e-08
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
          controlPID 8 2 0.5
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.04500995778379948 0.02501338076128503 -2.467192854967864e-05
        rotation 0.999999320349919 0.0010052735011066647 -0.0005905293285751098 4.636837219387676
        children [
          DEF WHEEL Transform {
            translation -0.007 0.01 -0.05
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0 0 0
                  transparency 1
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
            rotationStep 0.25
          }
        ]
        name "right wheel(1)"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
        rotationStep 0.25
      }
    }
    DEF lefthinge HingeJoint {
      jointParameters HingeJointParameters {
        position -8.299657739673409e-09
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
          controlPID 8 0 0
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation -0.04501012942703829 0.02497767197765864 -4.573397634734988e-05
        rotation -0.999998211339496 0.00047275269647763495 -0.0018313445051405448 1.2425810119893155
        children [
          DEF WHEEL Transform {
            translation 0.007 0 -0.05
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.1 0.1 0.1
                  transparency 1
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
            rotationStep 0.25
          }
        ]
        name "left wheel(1)"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    DEF front_ds DistanceSensor {
      translation 0.045 0 -0.01
      rotation 0 1 0 0
      name "front_ds"
      recognitionColors [
        0 0 0
      ]
      lookupTable [
        0 0 0
        1.27 1.27 0
      ]
    }
    DEF right_ds DistanceSensor {
      translation 0.01 -0.035 -0.01
      rotation 0 0 1 -1.57
      name "right_ds"
      lookupTable [
        0 0 0
        1.27 1.27 0
      ]
    }
    DEF left_ds DistanceSensor {
      translation 0.01 0.035 -0.01
      rotation 0 0 1 1.57
      name "left_ds"
      lookupTable [
        0 0 0
        1.27 1.27 0
      ]
    }
    Camera {
      translation 0.03 0 -0.045
      name "camera1"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
  ]
}
